### PR TITLE
[validate] - fixed an issue with validate -g

### PR DESCRIPTION
**Pull request by GuyAfik**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-8256

## Description
Fixed an issue where the **validate -g** failed when updating an empty `.pack-ignore` file when trying to retrieve its content from the local default branch.

### CODE REVIEW

File: demisto_sdk/commands/validate/validate_manager.py [hunk 1]
```diff
@@ -1994,11 +1994,15 @@ def get_all_files_edited_in_pack_ignore(self, modified_files: set) -> set:
...
-                f"{GitUtil.find_primary_branch(self.git_util.repo)}:{old_file_path}"
-            )
+            old_pack_ignore_content = get_remote_file(old_file_path, "master")
+            if not old_pack_ignore_content:
+                try:
+                    old_pack_ignore_content = self.git_util.get_local_remote_file_content(
+                        f"{GitUtil.find_primary_branch(self.git_util.repo)}:{old_file_path}"
+                    )
+                except GitCommandError:
+                    old_pack_ignore_content = ""
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:2004]: add a log to the `except` block. Not sure if this is `info` or `warning` though

Comment (GuyAfik) [reply demisto_sdk/commands/validate/validate_manager.py:2009]: ill do it warning

Comment (dorschw) [reply demisto_sdk/commands/validate/validate_manager.py:2012]: Tell the users how it affects them - is this ok? should it be fixed? What are the implications?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 ## Unreleased
 * Fixed an issue where logging arguments were not in the standard kebab-case. The new arguments are: **console-log-threshold**, **file-log-threshold**, **log-file-path**.
    Added a validation to ensure there's a '.' in the end of each description field in integration and script yml files.
+* Fixed an issue where the **validate -g** failed when updating an empty `.pack-ignore` file when trying to retrieve its content from the local default branch.
```
Comment (dorschw) [inline CHANGELOG.md:5]:
```suggestion
* Fixed an issue where the **validate -g** failed reading a `.pack-ignore` file that was previously empty.
```

File: demisto_sdk/commands/validate/validate_manager.py (hunk 1 above)
Comment (GuyAfik) [reply demisto_sdk/commands/validate/validate_manager.py:2013]: done

File: demisto_sdk/commands/validate/validate_manager.py [hunk 3]
```diff
@@ -1997,11 +1997,29 @@ def get_all_files_edited_in_pack_ignore(self, modified_files: set) -> set:
...
             # if the repo does not have remotes, get the .pack-ignore content from the master branch in Github
             # if the repo is not in remote / file cannot be found, try to take it from the latest commit on the default branch (usually master/main)
-            old_pack_ignore_content = get_remote_file(
-                old_file_path, "master"
-            ) or self.git_util.get_local_remote_file_content(
-                f"{GitUtil.find_primary_branch(self.git_util.repo)}:{old_file_path}"
-            )
+            old_pack_ignore_content = get_remote_file(old_file_path, "master")
+            if old_pack_ignore_content == b"":
+                # the .pack-ignore is empty in the remote
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:2002]:
```suggestion
            if old_pack_ignore_content == b"": # found as empty file in remote
```

**Review by dorschw — APPROVED** [review]

**Review by dorschw — APPROVED** [review]

**Pull request merged**